tooling/minimize/intrinsics/src/lib.rs

```diff
@@ -29,3 +34,112 @@ pub unsafe fn deallocate(ptr: *mut u8, size: usize, align: usize) {
     let layout = Layout::from_size_align(size, align).unwrap();
     unsafe { System.deallocate(ptr, layout); }
 }
+
+static THREAD_HANDLES: Mutex<Vec<Option<JoinHandle<()>>>> = Mutex::new(Vec::new());
+
+// Spawn returns the index+1 of the thread because we don't have a JoinHandle for thread 0.
+pub fn spawn<F>(f: F) -> usize
+where F: FnOnce() -> () + Send + 'static,
```

Comment:
Your current implementation will only actually work with `f: fn()`. That's also all you test. So please adjust the type signature for now.

But really this should have a signature more like `pthread_create`: `fn spawn(f: fn(*const ()), data: *const ())`. Then one can implement a wrapper around it that actually works for all `f: impl FnOnce()`.